### PR TITLE
Add pact provider for ocr

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1278

### Change description ###

Add pact provider for ocr
This way is the best way I can find. open for suggestions especially about pipeline.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        doNothing().when(authService).assertIsAllowedService(any());
+    }
+
+    @TestTemplate
+    @ExtendWith(PactVerificationInvocationContextProvider.class)
+    void pactVerificationTestTemplate(PactVerificationContext context) {
+        context.verifyInteraction();
+    }
+
+}
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java:48]: please update your IDE settings

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @LocalServerPort
+    private int port;
+
+    @MockBean
+    private AuthService authService;
+
+    @BeforeEach
+    void setupTestTarget(PactVerificationContext context) {
+        System.getProperties().setProperty("pact.verifier.publishResults", "true");
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java:36]: is there another way?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java (hunk 1 above)
Comment (mdayican) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java:48]: about ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java (hunk 2 above)
Comment (mdayican) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationContractTest.java:36]: could not find.

**Comment by rkondratowicz** [comment]
please fix codacy issues

**Comment by mdayican** [comment]
> please fix codacy issues

fixed

**Pull request merged**